### PR TITLE
Removes _lastUpdated and _id from search param tables

### PR DESCRIPTION
**Pull request by johnstairs**
## Description

1. Store the resource type ID in all search param and compartment assignment table. It will not be an index column, but it is available for filtering on. 
2. Do not store the resource id as in the TokenSearchParam table as the `_id` search param. It is already on the Resource table. The same for `_lastUpdated` in the DateTimeSearchParam. 

## Related issues
Addresses [AB#69044](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/69044)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/TokenSearchParameterRowGenerator.cs [hunk 1]
```diff
@@ -3,21 +3,26 @@
...
+
         public TokenSearchParameterRowGenerator(SqlServerFhirModel model)
             : base(model)
         {
         }
 
         internal override bool TryGenerateRow(short searchParamId, TokenSearchValue searchValue, out V1.TokenSearchParamTableTypeRow row)
         {
-            if (string.IsNullOrWhiteSpace(searchValue.Code))
+            // don't store if the code is empty or if this the Resource _id parameter. The id is already maintained on the Resource table.
```
Comment (cunninghamjc) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/TokenSearchParameterRowGenerator.cs]: Typo: "if this the Resource" -> "if this is the Resource"

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/SearchParameterRowGenerator.cs [hunk 2]
```diff
@@ -15,18 +16,24 @@ internal abstract class SearchParameterRowGenerator<TSearchValue, TRow> : ITable
...
         private readonly bool _isConvertSearchValueOverridden;
+        private readonly bool _isInitializeOverridden;
+        private bool _isInitialized;
 
         protected SearchParameterRowGenerator(SqlServerFhirModel model)
         {
             EnsureArg.IsNotNull(model, nameof(model));
             Model = model;
             _isConvertSearchValueOverridden = GetType().GetMethod(nameof(ConvertSearchValue), BindingFlags.Instance | BindingFlags.NonPublic).DeclaringType != typeof(SearchParameterRowGenerator<TSearchValue, TRow>);
+            _isInitializeOverridden = GetType().GetMethod(nameof(Initialize), BindingFlags.Instance | BindingFlags.NonPublic).DeclaringType != typeof(SearchParameterRowGenerator<TSearchValue, TRow>);
```
Comment (cunninghamjc) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/SearchParameterRowGenerator.cs]: Can you add a comment about why we're doing this?

**Comment by cunninghamjc** [comment]
> CREATE NONCLUSTERED INDEX IX_StringSearchParam_SearchParamId_TextWhereNoOverflow

Wrong index name?

---
Refers to: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/1.sql:489 in 68d685c. [](commit_id = 68d685cdbbe49f1e9dc643a40dbf063b50297f17, deletion_comment = False)

**Comment by cunninghamjc** [comment]
> CREATE NONCLUSTERED INDEX IX_StringSearchParam_SearchParamId_TextWhereNoOverflow

Wrong index name?

---
Refers to: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/1.sql:489 in 68d685c. [](commit_id = 68d685cdbbe49f1e9dc643a40dbf063b50297f17, deletion_comment = False)

**Comment by johnstairs** [comment]
> > CREATE NONCLUSTERED INDEX IX_StringSearchParam_SearchParamId_TextWhereNoOverflow
> 
> Wrong index name?
> 
> Refers to: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/1.sql:489 in [68d685c](https://github.com/microsoft/fhir-server/commit/68d685cdbbe49f1e9dc643a40dbf063b50297f17). [](commit_id = [68d685c](https://github.com/microsoft/fhir-server/commit/68d685cdbbe49f1e9dc643a40dbf063b50297f17), deletion_comment = False)

Fixing in #474

File: src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/SearchParameterRowGenerator.cs (hunk 2 above)
Comment (johnstairs) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/SearchParameterRowGenerator.cs]: I think this was a case of pointless optimizations. I'll remove.

|          Method |     Mean |     Error |    StdDev | Rank |
|---------------- |---------:|----------:|----------:|-----:|
|          Normal | 1.127 ns | 0.0582 ns | 0.0988 ns |    1 |
| MoreComplicated | 1.100 ns | 0.0247 ns | 0.0231 ns |    1 |

**Pull request merged**